cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+
+#### Examples
+##### Example 1
+
+* Cloud Provider deny tuples 
+-–deny-tuples=[“169.254.128.5;80;;]
+
+* Network Policy/Cilium Network Policy 
+None
+
+* Action
+Egress - Default Allow for all. Block access to 169.254.128.5:80 for all
+Ingress - Default Allow for all
+
+##### Example 2
+
+* Cloud Provider deny tuples 
+```-–d-eny-tuples=[“169.254.128.5;80;;]```
+
+* Network Policy/Cilium Network Policy 
+```
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: "l3-egress"
+spec:
+  endpointSelector:
+    matchLabels:
+      role: backend
+  egress:
+  - toEndpoints:
+    - matchLabels:
+        role: frontend
+```
+* Action
+Egress - Allow for all except for pods with label “role: backend”. For those pods, egress allowed only to pods with “role: frontend”. Block access to 169.254.128.5:80 for all
+Ingress - Default Allow for all
+
+
+#### Alternate Options
+#### Option 2 (CCNP CRD with new field)
+The proposal suggests introducing a new field called "defaultAllowEgress" in the ClusterwideCiliumNetworkPolicy (CCNP). This field aims to enforce a default allow instead of the default deny for egress traffic within the cluster. By default, the defaultAllowEgress field is disabled, indicating that the default deny is in effect. The scope of the defaultAllowEgress field is at the policy level, meaning it applies to the entire policy, even if there are multiple rules defined within the same policy. Default deny for egress traffic will take precedence if egress enabled in atleast one of Network Policy or Cilium Network Policy or CCNP that does not have the defaultAllowEgress flag set. 
+When creating or updating a policy, Cilium triggers the regeneration of the associated endpoint and its policies. It selects all rules that match the endpoint’s identity and in that process, it identifies if egress or ingress policy is enabled. It adds a default allow rule for egress traffic with identity 0 when the egress policy is not enabled. It skips adding this rule when egress policy is enabled. With the proposed change, Cilium will add a new field say  “DefaultAllowEgress” and will be set only if there exists CCNP with defaultAllowEgress set. If there are any other network policies or CCNP without that field, then it will be disabled. With this change, Cilium will add default allow rule for egress traffic if it satisfies any of these two conditions:
+
+- The egress policy is not enabled for the endpoint.
+- The egress policy is enabled and the DefaultAllowEgress field is set.
+
+```
+apiVersion: "cilium.io/v2"
+kind: CiliumClusterwideNetworkPolicy
+metadata:
+  name: "block-azure-wireserver"
+spec:
+  endpointSelector: {}
+  egressDeny:
+  - toCIDR:
+    - 169.254.128.5
+    toPorts:
+    - ports:
+      - port: "80"
+        protocol: "TCP"
+  defaultAllowEgress: true
+```
```

Comment:
This example is presented without context. It's good to see the proposal in context of the policy, but without wording that describes what it does, it's hard to reason about the semantics.